packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -326,6 +334,38 @@ describe('CDK Include', () => {
     );
   });
 
+  test('when a parameter in an Fn::Sub expression is substituted with a deploy-time value, it adds a new key to the Fn::Sub map', () => {
+    const parameter = new core.CfnParameter(stack, 'AnotherParam');
+    includeTestTemplate(stack, 'fn-sub-parameters.json', {
+      parameters: {
+        'MyParam': parameter.valueAsString,
```

Comment:
> it caused incorrect CloudFormation to be generated (an Fn::Join as an argument to Fn::Sub, which is not allowed).

I don't exactly know what this part of the PR description means, but it sounds foreboding.

Can we also test:

```
  parameters: {
    'MyParam': `its_a_${parameter.valueAsString}_concatenation`
  }
```

?

Does that do the right thing?

Reply:
Sure. Changed the test to do a concatenation.